### ISSUE TITLE
[Win 7] cpp-test crash when clicking the next button in the clipping example

### ISSUE THREAD

**Issue by meiry** [cocos2d/cocos2d-x#10054]
windows 7 32 bit visual studio 2012 crash when entering the clipping test and pressing the next button
error:

```
Unhandled exception at 0x007DFCA4 in cpp-tests.exe: 0xC0000005: Access violation reading location 0x00000000. 
```

**Comment by minggo** [comment]
What's the engine version?

**Comment by meiry** [comment]
Latest 3.4 rc1

**Comment by ghost** [comment]
We have test v3.4rc1 on our PC but we can't reproduce the crash(issue 10054-56) you said. Our test environment is win8.1 64bit vs2013update3 and win7 64bit vs2012. So can you please clean your rep and project and test again? Thank you!

**Comment by meiry** [comment]
sure i will do it , 
i by the way testing in 32bit  not 64bit  bit with 2012 express version.

**Comment by ghost** [comment]
Yes, I know that, but now we don't have the test environment as yours.If the crash still happens, we will try to setup the same environment to figure it out.

**Comment by ghost** [comment]
What I mean 64bit and 32bit is about the system, not the cocos2d-x project.

**Comment by meiry** [comment]
Ok i did little debugging , the problem remain by the way .
in function : 

```
void BaseClippingNodeTest::nextCallback(Ref* sender)
{
    Scene *s = new (std::nothrow) ClippingNodeTestScene();
    s->addChild(nextAction());
    Director::getInstance()->replaceScene(s);
    s->release();
}

when steeping into  s->release();

in :
```

void Ref::release()
{
    CCASSERT(_referenceCount > 0, "reference count should greater than 0");

the _referenceCount  is 2 
but when it out of the scop of this function things get messed up 

something strange is going on here :

```
static Layer* nextAction()
{
    sceneIdx++;
    sceneIdx = sceneIdx % MAX_LAYER;

    auto layer = (createFunctions[sceneIdx])();

    return layer;
}

where sceneIdx is always -1  so createFunctions doesn't return something 
```

**Comment by dabingnn** [comment]
It was fixed in pull request #10115

**Comment by meiry** [comment]
@minggo 
you can close it also its working great
